### PR TITLE
[flow-schema] apiVersion for FlowSchema and PriorityLevelConfiguration changed to the minimum available

### PR DESCRIPTION
**Pull request by vitaliy-sn**
## Description

apiVersion for FlowSchema and PriorityLevelConfiguration changed to the minimum available

## Why do we need it, and what problem does it solve?

In rare cases, when rolling back a k8s version, a situation arises that makes it impossible to apply the new flow-schema release

## Manual tests

```shell
# bootstrap 1.26
root@dev-master-0:~# kubectl version -o json | jq '.serverVersion.gitVersion'
"v1.26.14"
root@dev-master-0:~# kubectl -n d8-system get secret sh.helm.release.v1.flow-schema.v3 -o json | jq -r '.data.release' | base64 -d | base64 -d | gzip -d | jq -r '.manifest' | grep "^apiVersion: f" -A1
apiVersion: flowcontrol.apiserver.k8s.io/v1beta2
kind: FlowSchema
--
apiVersion: flowcontrol.apiserver.k8s.io/v1beta2
kind: PriorityLevelConfiguration


# update to 1.27
root@dev-master-0:~# kubectl version -o json | jq '.serverVersion.gitVersion'
"v1.27.11"
root@dev-master-0:~# kubectl -n d8-system get secret sh.helm.release.v1.flow-schema.v4 -o json | jq -r '.data.release' | base64 -d | base64 -d | gzip -d | jq -r '.manifest' | grep "^apiVersion: f" -A1
apiVersion: flowcontrol.apiserver.k8s.io/v1beta3
kind: FlowSchema
--
apiVersion: flowcontrol.apiserver.k8s.io/v1beta3
kind: PriorityLevelConfiguration

# rollback to 1.26
root@dev-master-0:~# kubectl version -o json | jq '.serverVersion.gitVersion'
"v1.26.14"
root@dev-master-0:~# kubectl -n d8-system get secret sh.helm.release.v1.flow-schema.v5 -o json | jq -r '.data.release' | base64 -d | base64 -d | gzip -d | jq -r '.manifest' | grep "^apiVersion: f" -A1
apiVersion: flowcontrol.apiserver.k8s.io/v1beta2
kind: FlowSchema
--
apiVersion: flowcontrol.apiserver.k8s.io/v1beta2
kind: PriorityLevelConfiguration

# switch d8 image from pr to main
sh.helm.release.v1.flow-schema.v4  sh.helm.release.v1.flow-schema.v5  sh.helm.release.v1.flow-schema.v6  
root@dev-master-0:~# kubectl -n d8-system get secret sh.helm.release.v1.flow-schema.v6 -o json | jq -r '.data.release' | base64 -d | base64 -d | gzip -d | jq -r '.manifest' | grep "^apiVersion: f" -A1
apiVersion: flowcontrol.apiserver.k8s.io/v1beta3
kind: FlowSchema
--
apiVersion: flowcontrol.apiserver.k8s.io/v1beta3
kind: PriorityLevelConfiguration


# switch d8 image from main to pr
sh.helm.release.v1.flow-schema.v5  sh.helm.release.v1.flow-schema.v6  sh.helm.release.v1.flow-schema.v7  
root@dev-master-0:~# kubectl -n d8-system get secret sh.helm.release.v1.flow-schema.v7 -o json | jq -r '.data.release' | base64 -d | base64 -d | gzip -d | jq -r '.manifest' | grep "^apiVersion: f" -A1
apiVersion: flowcontrol.apiserver.k8s.io/v1beta2
kind: FlowSchema
--
apiVersion: flowcontrol.apiserver.k8s.io/v1beta2
kind: PriorityLevelConfiguration
```

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: flow-schema
type: fix
summary: "apiVersion for `FlowSchema` and `PriorityLevelConfiguration` changed to the minimum available."
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/static'...

File: modules/011-flow-schema/templates/flowschema.yaml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {{- if gt (len .Values.flowSchema.internal.namespaces) 0 }}
 ---
-  {{- if semverCompare ">= 1.29" .Values.global.discovery.kubernetesVersion }}
+  {{- if semverCompare "> 1.29" .Values.global.discovery.kubernetesVersion }}
 apiVersion: flowcontrol.apiserver.k8s.io/v1
-  {{- else if semverCompare ">= 1.26" .Values.global.discovery.kubernetesVersion }}
+  {{- else if semverCompare "> 1.26" .Values.global.discovery.kubernetesVersion }}
 apiVersion: flowcontrol.apiserver.k8s.io/v1beta3
-  {{- else if semverCompare ">= 1.23" .Values.global.discovery.kubernetesVersion }}
+  {{- else if semverCompare "> 1.23" .Values.global.discovery.kubernetesVersion }}
```
Comment (RomanenkoDenys) [inline modules/011-flow-schema/templates/flowschema.yaml:7]: minimal version now is 1.25, why we use > 1.23 here?

File: modules/011-flow-schema/templates/prioritylevelconfiguration.yaml [hunk 2]
```diff
@@ -1,9 +1,9 @@
 ---
-  {{- if semverCompare ">= 1.29" .Values.global.discovery.kubernetesVersion }}
+  {{- if semverCompare "> 1.29" .Values.global.discovery.kubernetesVersion }}
 apiVersion: flowcontrol.apiserver.k8s.io/v1
-  {{- else if semverCompare ">= 1.26" .Values.global.discovery.kubernetesVersion }}
+  {{- else if semverCompare "> 1.26" .Values.global.discovery.kubernetesVersion }}
 apiVersion: flowcontrol.apiserver.k8s.io/v1beta3
-{{- else if semverCompare ">= 1.23" .Values.global.discovery.kubernetesVersion }}
+{{- else if semverCompare "> 1.23" .Values.global.discovery.kubernetesVersion }}
```
Comment (RomanenkoDenys) [inline modules/011-flow-schema/templates/prioritylevelconfiguration.yaml:6]: minimal version now is 1.25, why we use > 1.23 here?

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
some comments

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/static'...

**Review by name212 — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [7774](https://github.com/deckhouse/deckhouse/pull/7774) to the branch [release-1.58](https://github.com/deckhouse/deckhouse/tree/release-1.58) successful!

**Pull request merged**